src/MgvHasOffers.sol

```diff
@@ -97,8 +92,10 @@ contract MgvHasOffers is MgvCommon {
           nextOfferData.offer = nextOfferData.offer.prev(prevId);
         }
         if (prevId != 0 && nextId != 0) {
-          return (local, shouldUpdateBranch);
+          return (local, false);
         }
+
+        shouldUpdateBranch = shouldUpdateBranch && prevId == 0 && !bestBin.strictlyBetter(offerBin);
```

Comment:
I don't understand why we should update the branch if `bestBin == offerBin`? Doesn't that mean that `local` already has everything cached correctly?

(I know this wasn't changed here, but it struck me when reading the code)

Reply:
in this case, we are deleting the first offer of a bin, so the best bin might have become empty. In that case we need to execute the following if branch at the end of the `dislodgeOffer` function:

```
      if (shouldUpdateBranch) {
        local = local.binPosInLeaf(leaf.bestNonEmptyBinPos());
      }
```

Reply:
ah, so it handles the case where we're deleting the best offer?

Reply:
yes!